beacon-chain/casper/validator.go

```diff
@@ -137,17 +133,12 @@ func AreAttesterBitfieldsValid(attestation *pb.AggregatedAttestation, attesterIn
 	return true
 }
 
-// GetProposerIndexAndShard returns the index and the shardID of a proposer from a given slot.
-func GetProposerIndexAndShard(shardCommittees []*pb.ShardAndCommitteeArray, lcs uint64, slot uint64) (uint64, uint64, error) {
-	if lcs < params.CycleLength {
```

Comment:
Did this move somewhere else? The value needs to be subtracted by `CycleLength` to get the right index.

Reply:
Ops. I forgot it's from `last_state_recalc - CYCLE_LENGTH ... last_state_recalc - 1`. I'll add it back

Reply:
We should add it back in at `GetShardAndCommitteesForSlot`, not here.